db_test.go

```diff
@@ -537,141 +537,6 @@ func Test_DB_Filter_Block(t *testing.T) {
 	}
 }
 
-// ErrorBucket is an objstore.Bucket implementation that supports error injection.
-type ErrorBucket struct {
-	iter                      func(ctx context.Context, dir string, f func(string) error, options ...objstore.IterOption) error
-	get                       func(ctx context.Context, name string) (io.ReadCloser, error)
-	getRange                  func(ctx context.Context, name string, off, length int64) (io.ReadCloser, error)
-	exists                    func(ctx context.Context, name string) (bool, error)
-	isObjNotFoundErr          func(err error) bool
-	isCustomerManagedKeyError func(err error) bool
-	attributes                func(ctx context.Context, name string) (objstore.ObjectAttributes, error)
-
-	upload func(ctx context.Context, name string, r io.Reader) error
-	delete func(ctx context.Context, name string) error
-	close  func() error
-}
-
-func (e *ErrorBucket) IsAccessDeniedErr(err error) bool {
-	return err != nil
-}
-
-func (e *ErrorBucket) Iter(ctx context.Context, dir string, f func(string) error, options ...objstore.IterOption) error {
-	if e.iter != nil {
-		return e.iter(ctx, dir, f, options...)
-	}
-
-	return nil
-}
-
-func (e *ErrorBucket) Get(ctx context.Context, name string) (io.ReadCloser, error) {
-	if e.get != nil {
-		return e.get(ctx, name)
-	}
-
-	return nil, nil
-}
-
-func (e *ErrorBucket) GetRange(ctx context.Context, name string, off, length int64) (io.ReadCloser, error) {
-	if e.getRange != nil {
-		return e.getRange(ctx, name, off, length)
-	}
-
-	return nil, nil
-}
-
-func (e *ErrorBucket) Exists(ctx context.Context, name string) (bool, error) {
-	if e.exists != nil {
-		return e.exists(ctx, name)
-	}
-
-	return false, nil
-}
-
-func (e *ErrorBucket) IsObjNotFoundErr(err error) bool {
-	if e.isObjNotFoundErr != nil {
-		return e.isObjNotFoundErr(err)
-	}
-
-	return false
-}
-
-func (e *ErrorBucket) IsCustomerManagedKeyError(err error) bool {
-	if e.isCustomerManagedKeyError != nil {
-		return e.isCustomerManagedKeyError(err)
-	}
-
-	return false
-}
-
-func (e *ErrorBucket) Attributes(ctx context.Context, name string) (objstore.ObjectAttributes, error) {
-	if e.attributes != nil {
-		return e.attributes(ctx, name)
-	}
-
-	return objstore.ObjectAttributes{}, nil
-}
-
-func (e *ErrorBucket) Close() error {
-	if e.close != nil {
-		return e.close()
-	}
-
-	return nil
-}
-
-func (e *ErrorBucket) Upload(ctx context.Context, name string, r io.Reader) error {
-	if e.upload != nil {
-		return e.upload(ctx, name, r)
-	}
-
-	return nil
-}
-
-func (e *ErrorBucket) Delete(ctx context.Context, name string) error {
-	if e.delete != nil {
-		return e.delete(ctx, name)
-	}
-
-	return nil
-}
-
-func (e *ErrorBucket) Name() string { return "error bucket" }
-
-func Test_DB_OpenError(t *testing.T) {
```

Comment:
@thorfour seems like this test was not doing much so I removed it. Is that ok?

Reply:
It was validating that if the DB encountered an error when opening that it didn't panic and that we could successfully open the DB in subsequent attempts. 

Why remove it?